MaxDuino.ino

```diff
@@ -1213,25 +1217,31 @@ void downHalfSearchFile() {
 void seekFile() {    
   //move to a set position in the directory, store the filename, and display the name on screen.
   entry.close(); // precautionary, and seems harmless if entry is already closed
-  
-  while (!entry.open(&currentDir, currentFile, O_RDONLY))
+  if (dirEmpty)
   {
-    // if entry.open fails, when given an index, sdfat 2.1.2 automatically adjust curPosition to the next good entry
-    // (which means that we can just retry calling open, passing in curPosition)
-    // but sdfat 1.1.4 does not automatically adjust curPosition, so we cannot do that trick.
-    // We cannot call openNext, because (with sdfat 2.1.2) the position has ALREADY been adjusted, so you will actually
-    // miss a file.
-    // so need to do this manually (to support both library versions), via incrementing currentFile manually
-    currentFile++;
+    strcpy(fileName, PSTR("[EMPTY]"));
   }
+  else
+  {
+    while (!entry.open(&currentDir, currentFile, O_RDONLY) && currentFile<maxFile)
```

Comment:
I added this check too, currentFile<maxFile, to avoid any potential infinite loop